lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -210,15 +210,19 @@ export async function generateLockFile(
       commands.push(`yarn set version ${quote(yarnUpdate.newValue!)}`);
     }
 
-    if (process.env.HTTP_PROXY && !isYarn1) {
-      commands.push(
-        `yarn config set --home httpProxy ${quote(process.env.HTTP_PROXY)}`,
-      );
-    }
-    if (process.env.HTTPS_PROXY && !isYarn1) {
-      commands.push(
-        `yarn config set --home httpsProxy ${quote(process.env.HTTPS_PROXY)}`,
-      );
+    if (!process.env.RENOVATE_X_YARN_IGNORE_PROXY) {
+      if (process.env.HTTP_PROXY && !isYarn1) {
+        commands.push('yarn config unset --home httpProxy');
```

Comment:
what does the unset improve? 🤔

Reply:
Unset returns "nothing is set"

Reply:
Actually should add an || true here

Reply:
an `|| true` will maybe fail on windows?

Reply:
Do we have any safe way to ignore the result of such a command and continue?

Reply:
it seems we do this on other places too 😕 
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/modules/manager/composer/artifacts.ts#L172
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/modules/manager/npm/post-update/yarn.ts#L159
https://github.com/renovatebot/renovate/blob/c93cd9616fcf8509cd7e9efd3577a395d916d45a/lib/util/exec/containerbase.ts#L342

Reply:
Let's leave as-is in this PR